### ISSUE TITLE
Ray Client not connecting

### ISSUE THREAD

**Issue by al-rigazzi** [ray-project/ray#17662]
When trying to connect to the client (with either `ray.init("ray://<host_ip>:10001").connect()` or `ray.client("<host_ip>:10001").connect()`, I get the following Redis error (where `<host_ip>` and `<CONDA_ENV>` correctly reflect my system setup):

```python
Got Error from logger channel -- shutting down: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.UNKNOWN
	details = "Exception iterating responses: Server startup failed."
	debug_error_string = "{"created":"@1628351834.649740150","description":"Error received from peer ipv4:<host_ip>:10001","file":"src/core/lib/surface/call.cc","file_line":1067,"grpc_message":"Exception iterating responses: Server startup failed.","grpc_status":2}"
>
Traceback (most recent call last):
  File "<CONDAENV>/lib/python3.7/site-packages/ray/util/client/logsclient.py", line 42, in _log_main
    for record in log_stream:
  File "<CONDAENV>/lib/python3.7/site-packages/grpc/_channel.py", line 426, in __next__
    return self._next()
  File "<CONDAENV>/lib/python3.7/site-packages/grpc/_channel.py", line 826, in _next
    raise self
grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.UNKNOWN
	details = "Exception iterating responses: Server startup failed."
	debug_error_string = "{"created":"@1628351834.649740150","description":"Error received from peer ipv4:<host_ip>:10001","file":"src/core/lib/surface/call.cc","file_line":1067,"grpc_message":"Exception iterating responses: Server startup failed.","grpc_status":2}"
>

---------------------------------------------------------------------------
ConnectionAbortedError                    Traceback (most recent call last)
<ipython-input-33-c378e9afda83> in <module>

----> 6 ray.client("<host_ip>:10001").connect()

<CONDAENV>/lib/python3.7/site-packages/ray/client_builder.py in connect(self)
     99         """
    100         client_info_dict = ray.util.client_connect.connect(
--> 101             self.address, job_config=self._job_config)
    102         dashboard_url = ray.get(
    103             ray.remote(ray.worker.get_dashboard_url).remote())

<CONDAENV>/lib/python3.7/site-packages/ray/util/client_connect.py in connect(conn_str, secure, metadata, connection_retries, job_config, namespace, ignore_version)
     34         connection_retries=connection_retries,
     35         namespace=namespace,
---> 36         ignore_version=ignore_version)
     37 
     38 

<CONDAENV>/lib/python3.7/site-packages/ray/util/client/__init__.py in connect(self, conn_str, job_config, secure, metadata, connection_retries, namespace, ignore_version)
     81                 connection_retries=connection_retries)
     82             self.api.worker = self.client_worker
---> 83             self.client_worker._server_init(job_config)
     84             conn_info = self.client_worker.connection_info()
     85             self._check_versions(conn_info, ignore_version)

<CONDAENV>/lib/python3.7/site-packages/ray/util/client/worker.py in _server_init(self, job_config)
    523                 init_req = ray_client_pb2.InitRequest(
    524                     job_config=pickle.dumps(job_config))
--> 525                 self._call_init(init_req)
    526                 runtime_env.upload_runtime_env_package_if_needed(job_config)
    527                 runtime_env.PKG_DIR = old_dir

<CONDAENV>/python3.7/site-packages/ray/util/client/worker.py in _call_init(self, init_request)
    504         if not init_resp.ok:
    505             raise ConnectionAbortedError(
--> 506                 f"Init Failure From Server:\n{init_resp.msg}")
    507         return
    508 

ConnectionAbortedError: Init Failure From Server:
Traceback (most recent call last):
  File "<CONDAENV>/python3.7/site-packages/ray/util/client/server/proxier.py", line 426, in Datapath
    client_id, job_config):
  File "<CONDAENV>/python3.7/site-packages/ray/util/client/server/proxier.py", line 200, in start_specific_server
    output, error = self.node.get_log_file_handles(
  File "<CONDAENV>/python3.7/site-packages/ray/util/client/server/proxier.py", line 168, in node
    connect_only=True)
  File "<CONDAENV>/python3.7/site-packages/ray/node.py", line 181, in __init__
    session_name = _get_with_retry(redis_client, "session_name")
  File "<CONDAENV>/python3.7/site-packages/ray/node.py", line 41, in _get_with_retry
    result = redis_client.get(key)
  File "<CONDAENV>/site-packages/redis/client.py", line 1606, in get
    return self.execute_command('GET', name)
  File "<CONDAENV>/lib/python3.7/site-packages/redis/client.py", line 898, in execute_command
    conn = self.connection or pool.get_connection(command_name, **options)
  File "/<CONDAENV>/lib/python3.7/site-packages/redis/connection.py", line 1192, in get_connection
    connection.connect()
  File "<CONDAENV>/lib/python3.7/site-packages/redis/connection.py", line 567, in connect
    self.on_connect()
  File "<CONDAENV>/lib/python3.7/site-packages/redis/connection.py", line 643, in on_connect
    auth_response = self.read_response()
  File "<CONDAENV>/lib/python3.7/site-packages/redis/connection.py", line 756, in read_response
    raise response
redis.exceptions.ResponseError: WRONGPASS invalid username-password pair
```

This has been a problem starting with Ray 1.4 and is a problem in Ray 1.5.1, but it was totally OK in Ray 1.3

**Comment by DmitriGekhtman** [comment]
Thanks for raising this here.

Recalling context from Slack -- this cluster is deployed manually by running Ray start on the relevant nodes with a custom Redis password.
@ijrsvt this looks similar to another Redis password issue you resolved recently -- it had something to do with the proxier.

It's generally not recommended to use a custom Redis password to secure the Ray head's Redis. (You should secure the Ray cluster's network instead.)
Regardless, we should get to the bottom of this bug.

**Comment by al-rigazzi** [comment]
@DmitriGekhtman Thank you for your reply! So, just to make sure, I am just using the Redis password option for `ray start`. Do you think I should not use that argument, and leave the default password? I am not sure all of our users can secure the ray network, especially on large shared systems.

Anyhow, let me know if I can provide more details to help you solving the issue.

Thanks!

**Comment by DmitriGekhtman** [comment]
This is surprising -- I'm pretty sure https://github.com/ray-project/ray/pull/16885 made it into Ray 1.5.

**Comment by al-rigazzi** [comment]
It seems to be still an issue in 1.5.1. In fact, I tried not passing the `--redis_password` argument and everything works.

**Comment by ijrsvt** [comment]
@al-rigazzi Are you using `redis_password` or `redis-password` (the latter is the correct one). I was not able to reproduce this with:
```
ray start --head --redis-password='abc'
```
Then

```
import ray
ray.client("ray://localhost:10001").connect()
```

**Comment by ijrsvt** [comment]
Ah yep, `--redis_password` throws an error for me. 

Actually, I'm able to reproduce this on `1.5.1`, but not master.

**Comment by DmitriGekhtman** [comment]
Could

> Ah yep, `--redis_password` throws an error for me.
> 
> Actually, I'm able to reproduce this on `1.5.1`, but not master.

Would you mind doing a quick check with the 1.6.0 release branch?

**Comment by ijrsvt** [comment]
It is in the 1.6.0 release branch: 
https://github.com/ray-project/ray/blob/658e3668ae296be0348f05ed4c8a18ebcf64e226/python/ray/util/client/server/proxier.py#L221

**Comment by DmitriGekhtman** [comment]
and indeed not in the 1.5.1 branch
https://github.com/ray-project/ray/blob/releases/1.5.1/python/ray/util/client/server/proxier.py#L203

Just saw the fix in the release notes draft.

Glad we cleared it up: fixed in master and will be fixed in the upcoming Ray 1.6.0 release!

**Comment by al-rigazzi** [comment]
That's great to know! Thanks for looking into this, we look forward to the next release!
